streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java

```diff
@@ -494,8 +494,12 @@ public void postCommit(final boolean enforceCheckpoint) {
                 throw new IllegalStateException("Unknown state " + state() + " while post committing active task " + id);
         }
 
-        commitRequested = false;
+        clearCommitStatuses();
+    }
+
+    private void clearCommitStatuses() {
```

Comment:
Can we do a follow up PR that extends the `StreamTaskTest` unit test to verify that we reset the flags?

Reply:
done